src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts

```diff
@@ -94,27 +96,36 @@ export default class RadioCardGroup extends ComponentBase implements IComponent,
 	}
 
 	public get cards(): azdata.RadioCard[] {
-		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, azdata.RadioCard[]>((props) => props.cards, []);
+		return this.getSpecficProperties().cards ?? [];
 	}
 
 	public get cardWidth(): string | undefined {
-		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.cardWidth, undefined);
+		return this.getSpecficProperties().cardWidth ?? undefined;
 	}
 
 	public get cardHeight(): string | undefined {
-		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.cardHeight, undefined);
+		return this.getSpecficProperties().cardHeight ?? undefined;
 	}
 
 	public get iconWidth(): string | undefined {
-		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.iconWidth, undefined);
+		return this.getSpecficProperties().iconWidth ?? undefined;
 	}
 
 	public get iconHeight(): string | undefined {
-		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.iconHeight, undefined);
+		return this.getSpecficProperties().iconHeight ?? undefined;
 	}
 
 	public get selectedCardId(): string | undefined {
-		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.selectedCardId, undefined);
+		return this.getSpecficProperties().selectedCardId ?? undefined;
+	}
+
+	public get orientation(): string {
+		const x = this.getSpecficProperties().orientation ?? 'horizontal';
+		return x;
+	}
+
+	private getSpecficProperties(): azdata.RadioCardGroupComponentProperties {
```

Comment:
So I like the direction of this - but I'd rather see it done at the base component level if we can. We should be able to make Component generic and take in the properties type and use that for the getSpecificProperties call there instead.

Reply:
I'm still planning on doing the generics thing I was talking to you about, just not in this PR since it's kinda unrelated. I'll probably do that next/soon.